### PR TITLE
[DOCS] [Delta Lake] Add user guide for Delta Lake reads.

### PR DESCRIPTION
**Pull request by clarkzinzow**
This PR adds a user guide for Delta Lake reads.

### CODE REVIEW

File: docs/source/user_guide/integrations/delta_lake.rst [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+| `binary`                    | :meth:`daft.DataType.binary() <daft.DataType.binary>`                                    |
++-----------------------------+------------------------------------------------------------------------------------------+
+| **Nested Types**                                                                                                       |
++-----------------------------+------------------------------------------------------------------------------------------+
+| `struct(fields)`            | :meth:`daft.DataType.struct(fields) <daft.DataType.struct>`                              |
++-----------------------------+------------------------------------------------------------------------------------------+
+| `list(child_type)`          | :meth:`daft.DataType.list(child_type) <daft.DataType.list>`                              |
++-----------------------------+------------------------------------------------------------------------------------------+
+| `map(K, V)`                 | :meth:`daft.DataType.struct({"key": K, "value": V}) <daft.DataType.struct>`              |
```
Comment (jaychia) [inline docs/source/user_guide/integrations/delta_lake.rst:112]: We should update the map type to @colin-ho 's new dedicated Map type

**Review by jaychia — APPROVED** [review]

File: docs/source/user_guide/integrations/delta_lake.rst (hunk 1 above)
Comment (clarkzinzow) [reply docs/source/user_guide/integrations/delta_lake.rst:112]: Good catch!

File: docs/source/user_guide/integrations/delta_lake.rst [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+3. **Multi-cloud Support:** Daft supports reading Delta Lake tables from AWS S3, Azure Blob Store, and GCS, as well as local files.
+
+Installing Daft with Delta Lake Support
+***************************************
+
+Daft internally uses the `deltalake <https://pypi.org/project/deltalake/>`__ Python package to fetch metadata about the Delta Lake table, such as paths to the underlying Parquet files and table statistics. The ``deltalake`` package therefore must be installed to read Delta Lake tables with Daft, either manually or with the below ``getdaft[deltalake]`` extras install of Daft.
+
+.. code-block:: shell
+
+    pip install -U getdaft[deltalake]
```
Comment (samster25) [inline docs/source/user_guide/integrations/delta_lake.rst:19]:
```suggestion
    pip install -U "getdaft[deltalake]"
```

**Review by samster25 — APPROVED** [review]

File: docs/source/user_guide/integrations/delta_lake.rst (hunk 2 above)
Comment (clarkzinzow) [reply docs/source/user_guide/integrations/delta_lake.rst]: Ah good catch!

**Pull request merged**